keps/sig-cli/3805-ssa-default/README.md

```diff
@@ -0,0 +1,567 @@
+
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+
+Feature is client-side only, no.
+
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+Feature is client-side only, no.
+
+###### Does enabling the feature change any default behavior?
+
+No, enabling the `auto` feature means actively selecting it.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The feature can be disabled by unsetting the `KUBECTL_AUTO_SERVER_SIDE` environment variable.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Toggling the flag value can have impact, but that's already the case.
+Changing the value to an automatic detection will actually help with
+that.
+
+###### Are there any tests for feature enablement/disablement?
+
+Since enablement is not done through a feature gate, tests are fairly
+easy to implement.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+In the context of this change, a rollback would possibly be someone
+switching from server-side apply to client-side apply and vice-versa.
+This problem isn't new and is one of the reason that motivates this
+change. It mostly works well when the default fieldmanager is being
+used.
+
+<<[UNRESOLVED bad description of the problems here ]>>
+We could certainly make a much better analysis of the problems we have
+migrating from server-side to client-side and vice-versa here.
+<<[/UNRESOLVED]>>
+
+###### What specific metrics should inform a rollback?
+
+People should inform their decision based on the direct error they get
+from kubectl as they are trying to apply their resources.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+We have a series of tests for that.
+
+<<[UNRESOLVED find the tests ]>>
+I need to add the tests here.
+<<[/UNRESOLVED]>>
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+This will eventually change the default behavior of `kubectl apply` in
+possibly surprising way. `kubectl apply` can now fail because of a
+conflict, which it wouldn't before on newly created objects. Another
+surprising behavior is that since fields can be owned by multiple
+actors, removing a field will not necessarily remove the field from the
+cluster's resource.
+
+### Monitoring Requirements
+
+N/A. No monitoring in place.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+`kubectl apply -V8` can help identify what type of apply is used. Also
+checking the presence of the last applied annotation on the cluster
+resource indicates whether the resource was client-side or server-side
+applied.
```

Comment:
nit: the last part isn't strictly true since resources that went through the upgrade path using the default field manager will have their annotations sync'd by the server, right?

Reply:
Removed, added something else I thought about :-)